Clang scan-build issues

/home/beldmit/upstream/engine/test_sign.c:220:44: warning: Null pointer passed to 1st parameter expecting 'nonnull'
    T(EVP_PKEY_set_type_str(key1, algname, strlen(algname)));
                                           ^~~~~~~~~~~~~~~
/home/beldmit/upstream/engine/test_sign.c:23:23: note: expanded from macro 'T'
#define T(e) ({ if (!(e)) { \
                      ^
1 warning generated.
/home/beldmit/upstream/engine/benchmark/sign.c:93:16: warning: Null pointer passed to 1st parameter expecting 'nonnull'
                                data_len = atoi(optarg);
                                           ^~~~~~~~~~~~
/home/beldmit/upstream/engine/benchmark/sign.c:96:14: warning: Null pointer passed to 1st parameter expecting 'nonnull'
                                cycles = atoi(optarg);
                                         ^~~~~~~~~~~~
2 warnings generated.

--- a/benchmark/sign.c
+++ b/benchmark/sign.c
@@ -85,8 +85,6 @@ int main(int argc, char **argv)
 	opterr = 0;
 	while((option = getopt(argc, argv, "l:c:C")) >= 0)
 	{
-		if(option == ':') option = optopt;
-		if(optarg && (optarg[0] == '-')) { optind--; optarg = NULL; }
 		switch (option)
 		{
 			case 'l':

